src/main/java/de/tum/in/www1/artemis/domain/notification/SingleUserNotificationFactory.java

```diff
@@ -196,4 +201,101 @@ public static SingleUserNotification createNotification(TutorialGroup tutorialGr
         return notification;
     }
 
+    /**
+     * Creates an instance of SingleUserNotification for message replies.
+     *
+     * @param answerPost           to which the notification is related
+     * @param notificationType     type of the notification that should be created
+     * @param users                who should be notified or are related to the notification
+     * @param responsibleForAction the user who is responsible for the action that triggered the notification
+     * @return an instance of SingleUserNotification
+     */
+    public static SingleUserNotification createNotification(AnswerPost answerPost, NotificationType notificationType, Set<User> users, User responsibleForAction) {
+        var title = findCorrespondingNotificationTitleOrThrow(notificationType);
+        if (users.isEmpty()) {
+            throw new IllegalArgumentException("No users provided for notification");
+        }
+        SingleUserNotification notification;
+        if (notificationType == NotificationType.CONVERSATION_NEW_REPLY_MESSAGE) {
+            var user = users.stream().findAny().orElseThrow();
+            notification = new SingleUserNotification(user, title,
+                    "You have new reply in a message by " + responsibleForAction.getName() + " in course (" + answerPost.getPost().getConversation().getCourse().getTitle() + ").");
+            notification.setTransientAndStringTarget(createMessageReplyTarget(answerPost, answerPost.getPost().getConversation().getCourse().getId()));
+            notification.setAuthor(responsibleForAction);
+        }
+        else {
```

Comment:
I just returned the notification inside the if statement

Reply:
That's already a nice improvement, but generally, it's recommended to limit the nesting and indentation.
Also, reading is better because you have the exceptional case first.
It's also semantically better because the functionality comes **after** all exceptional cases were handled.